src/main/java/net/mcreator/ui/views/editor/image/ImageMakerView.java

```diff
@@ -207,31 +207,47 @@ public void save() {
 
 	public void saveAs() {
 		Image image = canvasRenderer.render();
-		Object[] options = TextureType.getTypes(false);
-		int n = JOptionPane.showOptionDialog(mcreator, L10N.t("dialog.image_maker.texture_kind"),
-				L10N.t("dialog.image_maker.texture_type"), JOptionPane.YES_NO_CANCEL_OPTION,
-				JOptionPane.QUESTION_MESSAGE, null, options, options[0]);
-		String namec = VOptionPane.showInputDialog(mcreator, L10N.t("dialog.image_maker.enter_name"),
-				L10N.t("dialog.image_maker.image_name"), null, new OptionPaneValidatior() {
-
-					@Override public ValidationResult validate(JComponent component) {
-						return new RegistryNameValidator((VTextField) component,
-								L10N.t("dialog.image_maker.texture_name")).validate();
-					}
-				});
-		if (namec != null && n != -1) {
-			File exportFile = mcreator.getFolderManager()
-					.getTextureFile(RegistryNameFixer.fix(namec), TextureType.getTextureType(n, false));
-
-			if (exportFile.isFile())
-				JOptionPane.showMessageDialog(mcreator, L10N.t("dialog.image_maker.texture_type_name_exists"),
-						L10N.t("dialog.image_maker.resource_error"), JOptionPane.ERROR_MESSAGE);
-			else
-				FileIO.writeImageToPNGFile(ImageUtils.toBufferedImage(image), exportFile);
-			this.image = exportFile;
-			this.name = this.image.getName();
-			refreshTab();
-		}
+

```

Comment:
With this new code, you have:

1.  Completely eliminated validation if file with such name already exists
2. A way to cancel UI without having to press X (cancel button)

Please resolve these two

Reply:
1. ![image](https://user-images.githubusercontent.com/38427877/175855076-0b7a7df0-9561-46b2-a3e6-4f175fa56bad.png)
What did you do to not have it? Personally, it works fine.

2. I will fix it. Sorry.

Reply:
1. I think I selected block and made name that already existed. I will try again and report

Reply:
THis is exactly what I've done.

Reply:
Sorry, I must have been doing something wrong, it indeed work right :)